foreshadow/metrics.py

```diff
@@ -0,0 +1,157 @@
+"""Metrics used across Foreshadow for smart decision making."""
```

Comment:
Will make a folder and export through __init__... -> registration shouldn't be needed because users can write their own smart transformers and will have access to any code they write. Metrics should just make their life a little easier.